### ISSUE TITLE
Cannot create child entry

### ISSUE THREAD

**Issue by chileverde** [mitchcapper/KPEntryTemplates#71]
I just downloaded KeePass and your plugin for the first time.

I created a template for credit cards following your model.

Then I added an entry in a group other than Templates. I click on Templates tab and then on Set Template Parent. A blank window opens: "Select Parent Template Entry". But the area which apparently should show the template name is blank except for the word "Title".

Perhaps the problem is where I placed the plugin. Your instructions said to place it in the program folder, which I take to mean the same folder as KeePass.exe. But there is also a folder called Plugins. Initially, I placed it in both folders, but then my new entry for creating the template had TWO tabs called Template.

Maybe I should start with a fresh "installation". If I do, what is the correct location for your plugin with KeePass 2.45?

I am using the portable version in a folder of my laptop's HD. I can delete all the program files and extract again from ZIP if that would help. Can I still use my DB? Perhaps delete the Templates group first?

**Comment by Wild13c** [comment]
Hi, do you see a key on the 1st line under title, it will be at the far left of the line if it is there?  If so select it and that is the template you created.  I think you can give it a name but I have not tried that.

**Comment by chileverde** [comment]
Wild13c,

Thanks for your reply.

Unfortunately, I do not see the key.

I just see "Title" and the rest of the huge area is blank.

I'm using the latest versions: 8.0 for the plugin and 2.45 for KeyPass (portable) for Windows. If you are also using these versions, please tell me which folder you have the plugin in. I might try starting over, but I'm still not clear if the plugin should be in the same folder as KeePass.exe or in the Plugins folder.

chileverde

**Comment by Wild13c** [comment]
See above for clarifications on program and locations.  Here is a slightly modified version of the install.



    Download from https://github.com/mitchcapper/KPEntryTemplates/releases

    Extract the plug-in (KPEntryTemplates.plgx) and place in the KeePass plugin directory

    Start KeePass and open a database

    In KeePass, click "Edit > Add Group" and name it something like "Templates", press Enter to finish adding the new group.

    Select "File > Database Settings..." and select the "Advanced" tab.

    In the "Templates" group set, set the "Entry templates group:" select the group you just created, and click "OK". This will close the database settings dialog and return you to the main application.

    In the main application, select your "Templates" folder and click "Edit > Add Entry..."

    On the "Add Entry" dialog, select the "Template" tab and click "Init As Template"

    Build your new template by adding whatever fields you desire

    Be sure to switch back to the "Entry" tab and enter the name of the template in the "Title" field.

    Once you are satisfied, click "OK" to save the template

    Your template is now ready to use when adding or editing entries

**Comment by chileverde** [comment]
Thanks, Wild13c. I put the plugin in the plugin folder. I ran a clean copy of KeePass.exe. I created a new DB, added a test entry, and then created a template before I imported my old passwords. This time it worked, and it continued to work after I imported old password.

Before I created a new entry from the template for one credit card. But before I tried it, I first installed the KeePassHTTP plugin in KeePass, and I installed KeePassHttp-Connector in Firefox. It seems to work for logging in to websites, as long as I have only one account. However, there are many sites for which I have more than one account. For those sites, after I give permission for all the logins, no user name or password is autofilled into the fields. I think I'll try the KeePassXC-Browser extension: it is written for KeePassXC, but I read that it works for KeePass. When I tried it with XC, it worked fine when I had more than one account for a website.

For credit cards, I copied most of the field names in Mitch's example. But when I tried "buying" something on a website, the plugin does not recognize the fields. I have to copy and paste each field from KeePass to the website. I guess I got spoiled with Abine Blur, which works great but is not open-source, and I thought I should change to a password manager that is open-source.

Anyway, I really appreciate your help getting this far.

chileverde

**Comment by Wild13c** [comment]
Chileverde, you have it working somewhat.  I have never had to add the HTTP plugins you described and I run FireFox also.   If I have to guess you created Mitch's exact template.  If so, you need to look at the std. ENTRY tab which are Keepass default fields.  Add these to your Credit Card Template.  Once that is done you can double click on the URL to take you to the credit card or for that matter what ever website you want to go to.  Then you click in the Username on the website and press Ctrl, Alt, A and it will automatically enter the Username & Password.  There are ways of addressing some sites that have additional login in requirements but this works for most sites.  I can help you get going with that once we get it to this point.

**Comment by chileverde** [comment]
I'm trying some things out, based on your suggestion.

You are right—KeePass will enter user name and password in the login fields of a website without a browser extension in the browser or a plugin in KeePass. I discovered why this was not happening for me when I studied the items in Tools | Options | Advanced. The relevant items (I checked an extra one or two) provide that Autotype is enabled when something in the entry matches the target window _title_. There is no item for match with the target URL. All the entries which I imported from Blur have URL, user name, and password. They do not have a title. So I added titles to a few: Hertz and Powells [Books]. And I changed the hot keys for autofill from Ctrl+Alt+A and Ctrl+Alt+Shift+A (I think that's what they were, and I use them for something else on my system) to Ctrl+Shift+O and Ctrl+Shift+P.

Now if I go to one of those sites and click on the username field and press Ctrl+Shift+O, KeePass displays the listing, and after I press ENTER, KeePass types in the user name and password.

O.K. Now I am trying a website where I have three accounts, so there are three entries in KeePass. I have added the title of the website to each entry, so they all have the same title, but different user names and passwords. After I clicked on the username field and pressed Ctrl+Shift+O, KeePass displayed a list of the three user names, I picked one, and KeePass logged me in.

So far, pretty good (except that I have to add titles to >300 entries).

I don't have more time now, but I will have to experiment with trying to get credit card info into those CC fields on a website. In the original version of your post above, which I received by e-mail, you referred to clicking on the URL in the entry in KeePass to go to a site in the browser. I found that if I click once on the entry for a credit card (not opening the entry) in the list of entries, its details appear at the bottom of the program window in an area with a gray background. That's where I can click on the URL to open the website, although I am not interested in that at this point, since I want to use the CC to pay for a purchase on another website. I can select and right-click and copy items in the gray area, such as the credit card number, which I could then paste into the appropriate field on the website, but I hope I can eventually automate that process.

I guess Blur spoiled me: it would usually recognize the fields for CC info, and my list of credit cards would pop up. I would click on the one I wanted to use, and Blur would fill in _all_ the CC fields. But Blur is closed-source, and I think I should move to an open-source PM like KeyPass.

I will experiment more with KeePass and KeePassXC when I have time.

Thanks again for your help.
